test/Microsoft.AspNet.Razor.Runtime.Test/TagHelpers/CaseSensitiveTagHelperAttributeComparer.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Internal.Web.Utils;
+
+namespace Microsoft.AspNet.Razor.Runtime.TagHelpers
+{
+    public class CaseSensitiveTagHelperAttributeComparer : IEqualityComparer<IReadOnlyTagHelperAttribute>
+    {
+        public readonly static CaseSensitiveTagHelperAttributeComparer Default =
+            new CaseSensitiveTagHelperAttributeComparer();
+
+        private CaseSensitiveTagHelperAttributeComparer()
+        {
+        }
+
+        public bool Equals(
+            [NotNull] IReadOnlyTagHelperAttribute attributeX,
+            [NotNull] IReadOnlyTagHelperAttribute attributeY)
+        {
+            return
+                attributeX == attributeY || 
```

Comment:
Added for the case: `null` == `null`, could change it to be `(attributeX == null && attributeY == null) ||` but thought might as well just do the better check.